### PR TITLE
SRCH-2185 Use zeitwerk

### PR DESCRIPTION
**Pull request by jmax-fearless**
Updated directory structure, module names, and eager load paths to
accomodate zeitwerk's preferences.

### CODE REVIEW

**Comment by jmax-fearless** [comment]
The CodeClimate complaints are existing code that it's re-flagged. Requesting approval without changing them.

File: config/application.rb [hunk 1]
```diff
@@ -26,9 +26,12 @@ class Application < Rails::Application
...
 
-    # Turn on the classic autoloader, because zeitwerk blows
-    # up. SRCH-2185 is the ticket to move asis to zeitwerk.
-    config.autoloader = :classic
+    config.eager_load_paths += Dir[config.root.join('lib').to_s + '/']
+    config.eager_load_paths += Dir[config.root.join('lib', 'tasks').to_s + '/']
+    config.eager_load_paths += Dir[config.root.join('lib', 'assets').to_s + '/']
+    config.eager_load_paths += Dir[config.root.join('lib', 'capistrano').to_s + '/']
+    config.eager_load_paths += Dir[config.root.join('lib', 'capistrano', 'tasks').to_s + '/']
+    config.eager_load_paths += Dir[config.root.join('app', 'workers').to_s + '/']
```
Comment (MothOnMars) [inline config/application.rb:34]: Per our discussion, these lines don't appear to be necessary. I *believe* we can nix these as well as the following (as everything within `app/` should be autoloaded by default):
```ruby
config.autoload_paths += Dir[config.root.join('app', 'workers', '**/').to_s]
```

**Review by MothOnMars — CHANGES_REQUESTED** [review]
Test with suggested changes to `config/application.rb`.

File: config/application.rb (hunk 1 above)
Comment (jmax-fearless) [reply config/application.rb]: Removed and re-tested locally; you are correct.

**Review by MothOnMars — APPROVED** [review]
LGTM

**Pull request merged**